modules/codegen/src/smithy4s/codegen/SmithyResources.scala

```diff
@@ -0,0 +1,61 @@
+package smithy4s.codegen
+
+/**
+  * This construct aims at adding metadata information to the classpath to let Smithy4s
+  * know about code that may have been generated in upstream modules.
+  *
+  * A "smithy4sGenerated" metadata holds a list of all namespaces that will have already
+  * been generated by smithy4s.
+  */
+private[smithy4s] object SmithyResources {
+
+  def produce(
+      resourceOutputFolder: os.Path,
+      localSmithyFiles: List[os.Path],
+      namespaces: List[String]
+  ): List[os.Path] = {
+
+    val smithyFolder = resourceOutputFolder / "META-INF" / "smithy"
+    val namespaceTrackingSmithyFiles = namespaces.map { ns =>
```

Comment:
would it be problematic to write on smithy file per module instead? If we have 2 modules, and each modules has N namespaces, we'll write 2 * N smithy file with a metadata entry with an array containing one element (namespace). Why not writing one file per module, with an array containing N elements (namespace)

Reply:
I chose to write one file per namespace because otherwise what would you name the aggregating file ? If you choose a fixed name, thinking "it's okay because there will only be one per module", then you put your users who build fat jars through the pain of having to set assembly strategies for that particular file, and it's not trivial to aggregate several smithy files into one. 

Then you could ask "why not write a manifest file then, that can be simply concatenated during assembly ?". Well for starters, it still puts the users through the hassle, and secondly, it means that we need to write bespoke code to load the manifest file in question. Even if it's not that hard to write, it's still annoying to have to do it when smithy is perfectly able to aggregate metadata in from several files without us having to do anything. 

Moreover, these files are not really meant to be seen by anyone, so it really doesn't matter if there are N of them. The prepending of "smithy4s" reduces the risk of collision with actual files from userland.

Reply:
After a thought, rendering the smithy manifest creates this problem as well. I'm reconsidering your ask, not for the reason you're putting forward (because that particular reason does not present a particularly good rationale from a technical standpoint)

Reply:
I did not think about the name nor the assembly issues, that's for sure!